frontends/web/src/contexts/WCWeb3WalletProvider.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, useEffect, useState } from 'react';
+import WCWeb3WalletContext from './WCWeb3WalletContext';
+import { IWeb3Wallet, Web3Wallet } from '@walletconnect/web3wallet';
+import { Core } from '@walletconnect/core';
+
+type TProps = {
+    children: ReactNode;
+  }
+
+export const WCWeb3WalletProvider = ({ children }: TProps) => {
+  const [web3wallet, setWeb3wallet] = useState<IWeb3Wallet>({} as IWeb3Wallet);
+  const [isWalletInitialized, setIsWalletInitialized] = useState(false);
+
+  useEffect(() => {
+    if (!isWalletInitialized) {
+      const initializeWeb3Wallet = async () => {
+        try {
+          const core = new Core({
+            projectId: '89733df088867a1a1bf644013addd6cc',
+          });
+
+          const wallet = await Web3Wallet.init({
+            core,
+            metadata: {
+              name: 'BitBoxApp',
+              description: 'BitBoxApp',
+              url: 'https://bitbox.swiss',
+              // TODO: Add BitBox logo
+              icons: []
+            }
+          });
+
+          setWeb3wallet(wallet);
+          setIsWalletInitialized(true);
+        } catch (err: unknown) {
+          console.log('Error for initializing', err);
+        }
+      };
+
+      if (!isWalletInitialized) {
+        initializeWeb3Wallet();
+      }
+    }
+  }, [isWalletInitialized, web3wallet]);
+
+
+  const pair = async (params: { uri: string }) => {
+    const { uri } = params;
+
+    const topic = getTopicFromURI(uri);
+    const hasEverBeenRejected = pairingHasEverBeenRejected(topic);
+    if (hasEverBeenRejected) {
+      throw new Error('Please use a new URI!');
+    }
+    await web3wallet?.core.pairing.pair({ uri });
+  };
+
+  const getTopicFromURI = (wcURI: string) => {
+    try {
+      // Split the URI by ':' and then by '@', and take the part before the '@'
+      return wcURI.split(':')[1].split('@')[0];
```

Comment:
I thought we already have something similar, but couldn't find it.

wdyt about putting into `frontends/web/src/routes/account/walletconnect/utils.ts` but now maybe this utils shouldn't be located under `src/routes/account` but move the whole module to `src/utils`? WDYT?